### PR TITLE
GH-3 - Sort search results by Date (ASC/DESC)

### PR DESCRIPTION
**Pull request by whopiyush**
- This PR adds functionality to sort the results by date.
- There is a new settings option called 'Sort Results By': 
image

- Before the feature/Default:
image

- Sort (Ascending):
image

- Sort (Descending):
image

### CODE REVIEW

**Review by amitraj2203 — COMMENTED** [review]
Thanks for the work, I've left some comments can you please review them.
Also have you tried running the tests after these changes? I think we need to cover it in the unit or integration tests especially for edge cases (e.g., invalid values, API errors).

File: inc/classes/class-search-engine.php [hunk 1]
```diff
@@ -74,6 +82,11 @@ public function get_search_results( $search_query, $page = 1, $posts_per_page =
 			$this->get_api_url()
 		);
 
+		// Add sort parameter if set.
+		if ( ! empty( $this->sort_by ) && 'default' !== $this->sort_by ) {
+			$request_url = add_query_arg( array( 'sort' => $this->sort_by ), $request_url );
```
[the remark this answers is not in the document]
Comment (whopiyush) [reply inc/classes/class-search-engine.php:87]: I've added sanitization for the fields in the settings page now!

File: inc/classes/class-settings.php [hunk 2]
```diff
@@ -144,5 +159,52 @@ public function cse_search_type_field_cb() {
...
+	}
+
+	/**
+	 * Sanitize sort_by setting value.
+	 *
+	 * @param string $value The value to sanitize.
+	 * @return string The sanitized value.
+	 */
+	public function sanitize_sort_by( $value ) {
+		$allowed_values = array( 'default', 'date:a', 'date:d' );
```
Comment (Anuj-Rathore24) [inline inc/classes/class-settings.php:187]: Please utilize function, `get_allowed_sort_values` for getting allowed values.

**Review by Anuj-Rathore24 — CHANGES_REQUESTED** [review]

File: inc/classes/class-settings.php (hunk 2 above)
Comment (whopiyush) [reply inc/classes/class-settings.php]: Done!
